apps/wallet/src/ui/app/hooks/useGasBudgetInMist.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { STATE_OBJECT } from '../staking/usePendingDelegation';
+import { useGetObject } from './useGetObject';
+
+/**
+ * Converts the gas units in Mist based on reference gas price of the current epoch.
+ * Wallet currently calculates all the estimated gas budget in gas units, we use this
+ * hook to convert it to mist.
+ */
+export function useGasBudgetInMist(gasBudgetUnits: number) {
+    const { data, isLoading, isError, isSuccess } = useGetObject(STATE_OBJECT);
```

Comment:
Can we just use `getReferenceGasPrice` on the provider?

Reply:
Yeah, did it like this to avoid any extra requests but there might be caching in the provider as well. Will change it thanks

Reply:
Yeah makes sense, I don't think that's bad thinking, although I think there's basically a dedicated API for every use of the system object at this point, so we can probably start breaking it apart and get the benefit of more granular caching + refetching.

Reply:
Done